oss_src/fileio/libhdfs_shim.cpp

```diff
@@ -384,6 +385,26 @@ extern  "C" {
     search_prefixes = {""};
     search_suffixes = {""};
     file_name = "libjvm.dylib";
+
+    // Run /usr/libexec/java_home to get the libjvm location
+    std::string libjvm_location = "";
+    std::string java_home_cmd = "/usr/libexec/java_home";
+    try {
+      graphlab::process p;
+      p.popen(java_home_cmd,
+              std::vector<std::string>(),
+              STDOUT_FILENO);
+      libjvm_location = p.read_from_child();
+      logstream(LOG_INFO) << "Obtain JAVA_HOME from " << java_home_cmd << ": " << libjvm_location << std::endl;
+    } catch (...) {
+      logstream(LOG_WARNING) << "Error running " << java_home_cmd << std::endl;
+      libjvm_location = "";
+    }
+    if (!libjvm_location.empty()) {
+      // Make this location to be searched first `/usr/libexec/java_home`/jre/lib/server/libjvm.dylib
+      search_prefixes.insert(search_prefixes.begin(), libjvm_location);
```

Comment:
ah. the bug report was ambiguous. it is not " `/usr/libexec/java_home`" or "jre/lib/server/libjvm.dylib".
where x = `/usr/libexec/java_home`
the path is "$x/jre/lib/server/libjvm.dylib"

See the shell script in the bug report